Fixed issue of merging block contents upon pressing backspace at start of second block with some content

aryan7081 opened the pull request:
Ref #7263

Inside the `joinWithPreviousBlock` function of `packages/volto-slate/src/blocks/Text/keyboard/joinBlocks.js`, pressing Backspace (Delete on mac) at the start of a block incorrectly merged the previous block's content into the current block, instead of merging the current block into previous one.

This happened because the implementation relied on `mergeSlateWithBlockBackward(editor, otherBlock)` which altered the current block's editor slate causing the wrong merge direction.

This fix removes the dependency and explicitly appends the current block's children (`editor.children`) into the previous block's value, then deletes the current block.

wesleybl reviewed (changes requested):
This adds a new line to the previous paragraph. The expectation here is that the second paragraph will be added to the end of the previous paragraph, without adding a new line.

aryan7081 commented:
@wesleybl Thanks, I investigated and found that `mergeSlateWithBlockBackward` wasn't being used anywhere, so I fixed the bug in that function. Now, `joinWithPreviousBlock` uses it instead of duplicating custom logic. Also, I have added the username to the log files.

wesleybl reviewed (approved):
LGTM

wesleybl commented:
@aryan7081 It's because this PR doesn't fix #7263. They are similar but not the same situations.

aryan7081 commented:
@wesleybl Okay, got it. Thanks for the clarification.

wesleybl commented:
@aryan7081 There appears to be an intermittent error in the `AuthToken.test.js` test unrelated to your PR:

https://github.com/plone/volto/actions/runs/18220108302/job/51877958704?pr=7373#step:5:1022

Would you like to investigate this and create a new PR?

wesleybl commented:
@aryan7081 Thanks!

Could you please make a backport to Volto 18?

Could you also try a fix for #7263? It looks similar to this one.

aryan7081 commented:
> @aryan7081 Thanks!
> 
> Could you please make a backport to Volto 18?
> 
> Could you also try a fix for #7263? It looks similar to this one.

@wesleybl Okay, I'll prepare a backport to Volto 18 and also look into #7263 
Thanks and Regards

davisagli reviewed (commented):
I finally had a chance to test this and it works well. Thank you for working on it!

I tried a different but similar use case: pressing Delete at the end of a text block that precedes another text block. This adds the text from the next block to the current one (expected) but does not delete the next block (so you end up with duplicate text). Is it possible to adapt your work here to handle merging the blocks in that case as well? (Edit: I see that is the issue that is already discussed in https://github.com/plone/volto/issues/7263)

aryan7081 commented:
@davisagli Working on it.
Thanks and regards
